sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NotAssignedPrivateMember.cs

```diff
@@ -93,4 +94,14 @@ void M()
             b[0] = 42;
         }
     }
+
+    // https://github.com/SonarSource/sonar-csharp/issues/242
+    public class MyClass
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        private class InteropMethodArgument
+        {
+            public uint number; // Compliant, we don't raise on members of classes with StructLayout attribute
+        }
+    }
```

Comment:
Real corner case - should report on subclasses of _InteropMethodArgument_ since the attribute isn't inherited.
Not sure it's worth testing for though.

Reply:
We will raise in my opinion, but I am not going to add a test case :)